src/scanners/base.js

```diff
@@ -12,6 +12,12 @@ export default class BaseScanner {
     this._parsedContent = null;
     this._rulesProcessed = 0;
     this.options = {};
+
+    if (this.filename === undefined || this.filename === null ||
```

Comment:
Presumably if this.filename was `0` e.g. the number zero, then that would be invalid too because it should be a string. In which case this might be a good case to use a falsey check.

However, if you prefer this more explicit route  it would be good to use `typeof` for the undefined check.